lib/realtime_web/channels/realtime_channel/presence_handler.ex

```diff
@@ -67,15 +68,17 @@ defmodule RealtimeWeb.RealtimeChannel.PresenceHandler do
          payload,
          %{assigns: %{private?: true, policies: %Policies{presence: %PresencePolicies{write: nil}}}} = socket
        ) do
-    %{assigns: %{db_conn: db_conn, authorization_context: authorization_context}} = socket
+    %{assigns: %{authorization_context: authorization_context, tenant: tenant_id}} = socket
 
-    case run_authorization_check(socket, db_conn, authorization_context) do
-      {:ok, socket} ->
-        handle_presence_event("track", payload, socket)
+    with {:ok, db_conn} <- Connect.lookup_or_start_connection(tenant_id) do
```

Comment:
This with without else will return `{:error, _reason}` but I think the callers expect `{:error, socket}` instead